urwid/event_loop/trio_loop.py

```diff
@@ -142,8 +143,7 @@ def run(self) -> None:
 
         emulate_idle_callbacks = _TrioIdleCallbackInstrument(self._idle_callbacks)
 
-        # TODO(Aleksei): trio.MultiError is deprecated in favor of exceptiongroup package usage and `Except *`
-        with trio.MultiError.catch(self._handle_main_loop_exception):
+        with exceptiongroup.catch({BaseException: self._handle_main_loop_exception}):
```

Comment:
This change needs lower constraints for `trio` version

Reply:
Personally I prefer constraints on requirements, because support of unlimited legacy versions is an unlimited pain.

Reply:
Yup, I think this makes more sense too. I just wasn't sure what your preferences were. I've now pushed a version with a constraint on the earliest version of trio where this should work.